### PR TITLE
use helm for generating user deployable manifests

### LINKED ISSUE #850: [RFE]: Helm vs Jinja2
**Is your feature request related to a problem? Please describe.**
It's no fun [manually syncing Jinja2-generated manifests and helm chart files](https://github.com/kadalu/kadalu/pull/848#issuecomment-1157739688).

**Describe the solution you'd like**
Why not separate into:
* `templates/` directory contains only templated manifests rolled out by the operator (or kadalu controll plane)
* The rest goes into Helm. `manifests/` files can be generated via `helm template` from there.

**Describe alternatives you've considered**
Attempting to generate helm chart template files from jinja2. I don't see the bonus though. What's the benefit of having a jinja2 templated resource when it is not used in a programmatic python context?

**Additional context**
Just a suggestion to make it a bit easier to handle manifest generation.
I think it makes sense to differentiate between resources rolled out by Kadalu control plane (where jinja2 makes sense, haven't heard of a helm2 python library...) and the actual manifests deployed by users themselves.

### PR DESCRIPTION
**Pull request by leelavg**
Signed-off-by: Leela Venkaiah G <leelavg@thoughtexpo.com>

**What this PR does / why we need it**:
- distinction b/n user and operator deployable resources/manifests
- manifests generated by helm are placed in `manifests/` folder
- remove reduntant files which are no longer neede

**Which issue(s) this PR fixes**:
Fixes #850 

**Checklist**
- [ ] Documentation added
- [X] Tests updated
- [X] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

**Comment by leelavg** [comment]
- Quick PR to unblock #848 and #851 
- TODOs/Optimisation will be done later on

**Comment by leelavg** [comment]
@aravindavk seems cli tests are failing after binnacle update, could you pls take a look?

**Comment by aravindavk** [comment]
> @aravindavk seems cli tests are failing after binnacle update, could you pls take a look?

I will take a look. You can change the URL to use `0.4.1` till the fix is available. https://github.com/kadalu/kadalu/blob/devel/.github/workflows/on-pr-submit.yml#L79

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+- Use helm for generating devel manifests
```
Comment (aravindavk) [inline CHANGELOG.md:9]: Neat. :clap:

File: Makefile [hunk 2]
```diff
@@ -28,19 +28,45 @@ build-containers: cli-build
...
+		--set operator.kubernetesDistro=kubernetes \
+		--set operator.imagesHub=${IMAGES_HUB} \
+		--set operator.dockerUser=${DOCKER_USER} \
+		--set .Chart.version=${KADALU_VERSION} > manifests/kadalu-operator.yaml
+	@helm template --namespace kadalu helm/kadalu \
+		--set csi-nodeplugin.enabled=true \
+		--set csi-nodeplugin.kubernetesDistro=kubernetes \
+		--set csi-nodeplugin.imagesHub=${IMAGES_HUB} \
+		--set csi-nodeplugin.dockerUser=${DOCKER_USER} \
+		--set .Chart.version=${KADALU_VERSION} > manifests/csi-nodeplugin.yaml
```
Comment (aravindavk) [inline Makefile]: Can you make this as Make subcommand and then call `DISTRO=kubernetes $(MAKE) helm-manifest`

Comment (aravindavk) [reply Makefile]:
```makefile
DISTRO?=kubernetes
filename_suffix=
ifneq ($(DISTRO),kubernetes)
filename_suffix=-$(DISTRO)
endif

helm-manifest:
	@echo ---------------------------------------------------------------------
	@helm template --namespace kadalu helm/kadalu \
		--set operator.enabled=true \
		--set operator.kubernetesDistro=${DISTRO} \
		--set operator.imagesHub=${IMAGES_HUB} \
		--set operator.dockerUser=${DOCKER_USER} \
		--set .Chart.version=${KADALU_VERSION} > manifests/kadalu-operator${filename_suffix}.yaml
	@helm template --namespace kadalu helm/kadalu \
		--set csi-nodeplugin.enabled=true \
		--set csi-nodeplugin.kubernetesDistro=kubernetes \
		--set csi-nodeplugin.imagesHub=${IMAGES_HUB} \
		--set csi-nodeplugin.dockerUser=${DOCKER_USER} \
		--set .Chart.version=${KADALU_VERSION} > manifests/csi-nodeplugin${filename_suffix}.yaml
```

**Comment by leelavg** [comment]
> > @aravindavk seems cli tests are failing after binnacle update, could you pls take a look?
> 
> I will take a look. You can change the URL to use `0.4.1` till the fix is available. https://github.com/kadalu/kadalu/blob/devel/.github/workflows/on-pr-submit.yml#L79

- I see 0.4.4 also didn't fix the issue, looked at previous github actions and used 0.4.1 as you mentioned for now.

File: Makefile [hunk 3]
```diff
@@ -28,48 +29,59 @@ build-containers: cli-build
...
-	@IMAGES_HUB=${IMAGES_HUB} DOCKER_USER=${DOCKER_USER} KADALU_VERSION=${KADALU_VERSION} \
-		K8S_DIST=rke                                    \
-		python3 extras/scripts/gen_manifest.py manifests/kadalu-operator-rke.yaml
 	@echo
 	@echo "In the case of Rancher (RKE), deploy Kadalu Operator by running "
 	@echo "the following command"
 	@echo
-	@echo "kubectl apply -f manifests/kadalu-operator-rke.yaml"
-	@echo "kubectl apply -f manifests/csi-nodeplugin-rke.yaml"
+	@DISTRO=microk8s $(MAKE) -s helm-manifest
```
Comment (aravindavk) [inline Makefile]: rke?

**Comment by leelavg** [comment]
- addressed comments

**Comment by leelavg** [comment]
- the CI failure seems to be legitimate however isn't related to this PR
- in the [log](https://github.com/kadalu/kadalu/runs/6983652604?check_suite_focus=true#step:9:288), search for `cleaned` and could see a PVC was not deleted and moreover the pvc name seems a bit different. (some extra suffix with `data +%s`)

```
2022-06-21T12:04:38.4773562Z Not all PVCs are cleaned up properly
2022-06-21T12:04:38.8041972Z /mnt/storage-pool-4/virtblock/63
2022-06-21T12:04:38.8042600Z /mnt/storage-pool-4/virtblock/63/77
2022-06-21T12:04:38.8043657Z /mnt/storage-pool-4/virtblock/63/77/pvc-b4ed876a-289f-4d6e-b477-a5f8b635362d.1655812998.6117887
```

@aravindavk could you pls take a look if possible?

File: .github/workflows/on-pr-submit.yml [hunk 4]
```diff
@@ -76,7 +75,8 @@ jobs:
     # CLI Tests
     - name: Install Binnacle and build kubectl-kadalu
       run: |
-        curl -L https://github.com/kadalu/binnacle/releases/latest/download/binnacle -o binnacle
+        # TODO: revert to latest when it's stable
```
Comment (aravindavk) [inline .github/workflows/on-pr-submit.yml]: you can revert now.

**Review by aravindavk — APPROVED** [review]
Looks good.

File: .github/workflows/on-pr-submit.yml (hunk 4 above)
Comment (leelavg) [reply .github/workflows/on-pr-submit.yml]: ack

**Pull request merged**